include/optimizer.h

```diff
@@ -53,15 +54,22 @@ class Optimizer {
   *  \param index the unique index for the weight.
   *  \param weight the weight to update.
   *  \param grad gradient for the weight.
-  *  \param lr learning rate for this update.
   */
-  void Update(int index, NDArray weight, NDArray grad, mx_float lr);
+  void Update(int index, NDArray weight, NDArray grad);
   // TODO(zhangcheng-qinyinghua)
   // implement Update a list of arrays, maybe in the form of map
   //void Update(int index, std::vector<NDArray> weights, std::vector<NDArray> grad, mx_float lr);
 
+  /*!
+  *  \brief Serialize the optimizer parameters to a string.
+  *  \return serialization
+  */
+  std::string Serialize() const;
```

Comment:
The parameter assigned to optimizer will also be converted to string (see https://github.com/dmlc/MXNet.cpp/blob/202ec2daeb2fdd70c27b2134dc0dfe12cbf3d125/include/optimizer.h#L43), so byte encoding makes no difference here.